pkg/stdcopy/stdcopy_test.go

```diff
@@ -85,6 +85,24 @@ func TestWriteDoesNotReturnNegativeWrittenBytes(t *testing.T) {
 	}
 }
 
+func TestStdCopyWriteAndRead(t *testing.T) {
+	buffer := new(bytes.Buffer)
+	stdOutBytes := []byte(strings.Repeat("o", startingBufLen))
+	dstOut := NewStdWriter(buffer, Stdout)
+	dstOut.Write(stdOutBytes)
+	stdErrBytes := []byte(strings.Repeat("e", startingBufLen))
+	dstErr := NewStdWriter(buffer, Stderr)
+	dstErr.Write(stdErrBytes)
+	written, err := StdCopy(dstOut, dstErr, bytes.NewReader(buffer.Bytes()))
```

Comment:
True, but since we're working on the same buffer from dstOut and dstErr, we need to create a copy, otherwise we enter an infinite loop since we write to the same buffer. The other way would be to create a new dstOut and dstErr StdWriters here.